libs/platform/src/lib/components/list/list.component.ts

```diff
@@ -0,0 +1,631 @@
+    */
+    @Input()
+    set contentDensity(contentDensity: ContentDensity) {
+        this._contentDensity = contentDensity;
+        this.isCompact = contentDensity === 'compact';
+    }
+
+    /** setter and getter for _navigated */
+    get navigated(): boolean {
+        return this._navigated;
+    }
+
+    @Input('navigated')
+    set navigated(value: boolean) {
+        this._navigated = value;
+        this.itemEl.nativeElement.querySelector('ul').classList.add('fd-list--navigation');
+    }
+
+    /** setter and getter for _navigationIndicator */
+    get navigationIndicator(): boolean {
+        return this._navigationIndicator;
+    }
+
+    @Input('navigationIndicator')
+    set navigationIndicator(value: boolean) {
+        this._navigationIndicator = value;
+        this.itemEl.nativeElement.querySelector('ul').classList.add('fd-list--navigation-indication');
+    }
+
+    /** setter and getter for _hasByLine*/
+    get hasByLine(): boolean {
+        return this._hasByLine;
+    }
+
+    @Input('hasByLine')
+    set hasByLine(value: boolean) {
+        this._hasByLine = value;
+        this.itemEl.nativeElement.querySelector('ul').classList.add('fd-list--byline');
+    }
+
+    /** @hidden */
+    @Output()
+    selectedItemChange: EventEmitter<SelectionChangeEvent> = new EventEmitter<SelectionChangeEvent>();
+
+    /** @hidden */
+    constructor(protected _changeDetectorRef: ChangeDetectorRef,
+        public itemEl: ElementRef,
+        @Optional() @Self() public ngControl: NgControl,
+        @Optional() @Self() public ngForm: NgForm,
+        @Optional() @SkipSelf() @Host() formField: FormField,
+        @Optional() @SkipSelf() @Host() formControl: FormFieldControl<any>,
+        protected _listConfig?: ListConfig) {
+        super(_changeDetectorRef, ngControl, ngForm, formField, formControl);
+    }
+
+    /** @hidden */
+    /** Instailization of list with selection mode*/
+    ngOnInit(): void {
+
+        if (this.dsItems.length !== null && this.itemSize !== 0) {
+            this.startIndex = 0;
+            this.lastIndex = this.itemSize;
+            this.items = [];
+            this.items = this.dsItems.slice(this.startIndex, this.lastIndex);
+        } else {
+            this.items = this.dsItems;
+        }
+        this.id = `fdp-list-${nextListId++}`;
+        // using selection Model for multiselect
+        if (this.selectionMode === 'multi') {
+            this.multiSelect = true;
+            this.ariaMultiselectable = true;
+        } else { this.multiSelect = false; }
+        this.selectionModel = new SelectionModel<BaseListItem>(
+            this.multiSelect,
+            this.selectedItems
+        );
+
+        this.selectionModel.changed.subscribe(e => {
+            this.selectedItems = this.selectionModel.selected;
+            const event = new SelectionChangeEvent();
+            event.selectedItems = this.selectedItems;
+            this.selectedItemChange.emit(event);
```

Comment:
you should also emit event from stateChange (please see BaseInput)

Reply:
updated here and in all applicable places for list.componen.ts